protocol/v2/ssv/validator/non_committee_validator.go

```diff
@@ -106,47 +130,86 @@ func (ncv *CommitteeObserver) ProcessMessage(msg *queue.SSVMessage) error {
 	}
 
 	for key, quorum := range quorums {
-		role := ncv.getRole(msg, key.Root)
+		var operatorIDs []string
+		for _, share := range quorum {
+			operatorIDs = append(operatorIDs, strconv.FormatUint(share, 10))
+		}
+
 		validator, exists := ncv.ValidatorStore.ValidatorByIndex(key.ValidatorIndex)
 		if !exists {
 			return fmt.Errorf("could not find share for validator with index %d", key.ValidatorIndex)
 		}
-		MsgID := convert.NewMsgID(ncv.qbftController.GetConfig().GetSignatureDomainType(), validator.ValidatorPubKey[:], role)
-		if err := ncv.Storage.Get(MsgID.GetRoleType()).SaveParticipants(MsgID, slot, quorum); err != nil {
-			return fmt.Errorf("could not save participants %w", err)
-		} else {
-			var operatorIDs []string
-			for _, share := range quorum {
-				operatorIDs = append(operatorIDs, strconv.FormatUint(share, 10))
+
+		beaconRoles := ncv.getBeaconRoles(msg, key.Root)
+		if len(beaconRoles) == 0 {
+			logger.Warn("no roles found for quorum root",
+				zap.Uint64("validator_index", uint64(key.ValidatorIndex)),
+				fields.Validator(validator.ValidatorPubKey[:]),
+				zap.String("signers", strings.Join(operatorIDs, ", ")),
+				fields.BlockRoot(key.Root),
+				zap.String("qbft_ctrl_identifier", hex.EncodeToString(ncv.qbftController.Identifier)),
+			)
+		}
+
+		for _, beaconRole := range beaconRoles {
+			msgID := convert.NewMsgID(ncv.qbftController.GetConfig().GetSignatureDomainType(), validator.ValidatorPubKey[:], beaconRole)
+			roleStorage := ncv.Storage.Get(msgID.GetRoleType())
+			if roleStorage == nil {
+				return fmt.Errorf("role storage doesn't exist: %v", beaconRole)
+			}
+
+			existingQuorum, err := roleStorage.GetParticipants(msgID, slot)
+			if err != nil {
+				return fmt.Errorf("could not get participants %w", err)
+			}
+
+			if len(existingQuorum) > len(quorum) {
```

Comment:
`>=` ? 
do we want to override previous quorum of same amount? (override 1,2,3 with 1,2,4)

Reply:
though it should not happen..

Reply:
We don't need to override it, but now I think we need to merge quorums. If we receive 1,2,3,4,5 and then 2,3,4,5,6,7, then we'll lose 1. Pushed a fix